### PR TITLE
[CIAPP] Add test.parameters tag to TestNG tests

### PR DESCRIPTION
**Pull request by drodriguezhdez**
Add `test.parameters` tag to TestNG tests.

### CODE REVIEW

**Review by nachoBonafonte — APPROVED** [review]

File: dd-java-agent/instrumentation/testng-6.4/src/main/java/datadog/trace/instrumentation/testng/TestNGDecorator.java [hunk 1]
```diff
@@ -54,4 +58,22 @@ public void onTestIgnored(final AgentSpan span, final ITestResult result) {
...
     }
   }
+
+  private boolean hasParameters(final ITestResult result) {
+    return result.getParameters() != null && result.getParameters().length > 0;
+  }
+
+  // We build manually the JSON for test.parameters tag.
+  // Example: {"0":"param1","1":"param2"}
+  private String buildParametersTagValue(final ITestResult result) {
```
Comment (tonyredondo) [inline dd-java-agent/instrumentation/testng-6.4/src/main/java/datadog/trace/instrumentation/testng/TestNGDecorator.java:68]: Please note the output format for `test.parameters` contains an `arguments` submap with the name of the parameter + the value.  is like: 
image

**Review by tonyredondo — CHANGES_REQUESTED** [review]

File: dd-java-agent/instrumentation/testng-6.4/src/main/java/datadog/trace/instrumentation/testng/TestNGDecorator.java (hunk 1 above)
Comment (drodriguezhdez) [reply dd-java-agent/instrumentation/testng-6.4/src/main/java/datadog/trace/instrumentation/testng/TestNGDecorator.java:68]: Fixed!

Comment (tonyredondo) [reply dd-java-agent/instrumentation/testng-6.4/src/main/java/datadog/trace/instrumentation/testng/TestNGDecorator.java:68]: now you have the submap, but the key is an `index`, and the value is the parameter `name` isn't?

Should be the name and the actual parameter value. In my example the argument is called `secondsPassed` and the value of the argument for that execution was `60`.

Comment (drodriguezhdez) [reply dd-java-agent/instrumentation/testng-6.4/src/main/java/datadog/trace/instrumentation/testng/TestNGDecorator.java:68]: It's not possible to know the parameter name, just the value. That's why I use an index as best-effort approach.

Comment (tonyredondo) [reply dd-java-agent/instrumentation/testng-6.4/src/main/java/datadog/trace/instrumentation/testng/TestNGDecorator.java:68]: Oh ok, thanks for clarifying that.

**Review by tonyredondo — APPROVED** [review]

**Review by bantonsson — APPROVED** [review]

**Pull request merged**